lib/spreedly.ex

```diff
@@ -1,2 +1,99 @@
 defmodule Spreedly do
+  @moduledoc false
+  use HTTPoison.Base
+
+  alias HTTPoison.{Response, AsyncResponse, Error}
+  alias Spreedly.Environment
+
+  @spec get_request(Environment.t, String.t, Keyword.t, ((any) -> any)) :: {:ok, any} | {:error, any}
```

Comment:
I'm still not in the `@spec` camp because I feel like it adds so much noise and I haven't experienced the value yet. Perhaps this is a good library to try it to help convince me of the value.

Reply:
I'm joining the spec camp slowly, for this project being able to see that expected types in the doc without having to dig into the source is very nice. Dialyzer caught a few mistakes I had made that would have gone unnoticed - although it wasn't immediately apparent what the mistake was :(

Reply:
🆒 Will be interesting to see!